all/guide/passthrough.md

```diff
@@ -0,0 +1,198 @@
+---
+sidebarDepth: 2
+title: Passthrough migration
+---
+
+# Passthrough migration
+
+::::: row
+:::: left
+
+This guide describes the **passthrough** strategy for migrating password-based login from your existing system to Userfront.
+
+Using this approach, you get a standard set of client-side interactions for all new and existing users while you migrate your login to a new system.
+
+::::
+:::: right
+
+::: tip
+Userfront does not store user passwords in a recoverable format (it stores a [hash](https://en.wikipedia.org/wiki/Cryptographic_hash_function)), and this guide assumes that your system also does not store passwords in a recoverable format.
+:::
+
+::::
+:::::
+
+## How the passthrough works
+
+::::: row
+:::: left
+
+The first time an existing user attempts to log into your application with Userfront, we forward their request to your old login system using whatever format your old system uses.
+
+Based on the response from your old system, Userfront either logs the user in and stores their password hash for future use, or rejects the login along with a message based on your old system's response.
+
+::::
+:::: right
+
+<br/>
+
+<img src="https://res.cloudinary.com/component/image/upload/v1650036211/guide/passthrough-diagram.png" alt="Passthrough diagram" style="width: 500px;max-width:100%;">
+
+:::::
+
+## Passthrough steps
+
+::::: row
+:::: left
+
+To perform a successful password migration:
+
+1. <u>**Import your existing user data into Userfront**</u>, excluding passwords. You can use [CSV upload](/guide/import-export.html) for smaller data sets, or [contact us](mailto:team@userfront.com) for larger data sets.
+
+2. <u>**Set up a passthrough route**</u>. Userfront passes user login requests through to your old system and observes the responses in order to automatically generate a database of password hashes.
+
+3. <u>**Wait for the password transfer window**</u>. Wait as long as desired: typically 1-6 months, or until 90-100% of active users have transferred. As each user's password is verified, Userfront stores their password hash and can handle their subsequent logins directly.
+
+4. <u>**Disconnect the passthrough**</u>. Users who did not log in during the password transfer window are prompted to reset their password upon their next login.
```

Comment:
I would explicitly say "After you're finished transferring passwords, disconnect the passthrough"; it's obvious from context but saying it outright means the reader has to do less work.